packages/reactivity/src/ref.ts

```diff
@@ -72,6 +73,77 @@ class RefImpl<T> {
       trigger(toRaw(this), TriggerOpTypes.SET, 'value', newVal)
     }
   }
+
+  [Symbol.asyncIterator](): AsyncIterator<T> {
+    return new RefAsyncIterator(this)
+  }
+}
+
+/**
+ * Converts a RefImpl instance into an AyncIterator that can
+ * be used in a `for await` loop, or with other libraries like RxJS.
+ */
+class RefAsyncIterator<T> implements AsyncIterator<T> {
```

Comment:
Since this is probably going to be a pretty niche use case (in the context of all Composition API usage), ideally we want to make this part tree-shakeable.

Two ideas:

1. A separate `asyncIterableRef()` API
2. A `makeAsyncIterable(ref)` helper

There could also be arguments about whether this belongs in core. The obvious benefit of the current implementation is that any refs will be rx compatible - however the tradeoff is it adds core API surface for users who don't use Rx with Vue.

I'm thinking this might belong better as a utility function under vueuse... /cc @antfu

Reply:
RxJS compatibility is a bonus to this, really. This is more about being compatible with any API that could consume an asynciterator. In theory, you could build your own reactive combinators this way.  Something to think about.

If we only care about RxJS compat, and it needs to be tree-shakable, it would probably be more efficient to use something like what exists in [Vue Use](https://vueuse.org/rxjs/README.html).